engine/run.ts

```diff
@@ -12,47 +12,57 @@ class Run {
-                let engine = new Engine(key, value, this.playbooks[playbookIndex]);
+        try {
+            this.parseArgs();
+            this.parsePlaybooks();
+            this.parseEnvironments();
+            for (let entry of Array.from(this.environments.entries())) {
+                let key = entry[0];
+                let value = entry[1];
+                for (let playbookIndex in this.playbooks) {
+                    let engine = new Engine(key, value, this.playbooks[playbookIndex]);
 
-                for (let varEntry of Array.from(this.args.entries())) {
-                    engine.setVariable(varEntry[0], varEntry[1]);
-                }
+                    for (let varEntry of Array.from(this.args.entries())) {
+                        engine.setVariable(varEntry[0], varEntry[1]);
+                    }
 
-                try {
-                    await engine.run();
-                } catch (error) {
-                    console.error(error);
-                    errors.push(error);
+                    try {
+                        await engine.run();
+                    } catch (error) {
+                        console.error(error);
+                        errors.push(error);
+                    }
                 }
             }
+        } catch (error) {
+            console.error(error);
+            errors.push(error);
         }
+        
         if (errors.length != 0) {
             console.log("Errors", errors);
         }
         return errors.length == 0;
     }
 
     parsePlaybooks() {
-        let parser = new Parser();
+        try {
+            let parser = new Parser();
 
-        let playbooksDir = (<string>this.args.get("playbooksDir")) || __dirname + "/../playbooks/";
-        let playbookDirs = fs.readdirSync(playbooksDir);
-        for (let index in playbookDirs) {
-            let indexFile = playbooksDir + playbookDirs[index] + "/index.asciidoc";
-            if (fs.existsSync(indexFile)) {
-                let playbook = parser.parse(indexFile);
-                playbook.name = playbookDirs[index] + "/";
-                playbook.path = playbooksDir + playbookDirs[index] + "/";
-                this.playbooks.push(playbook);
+            let playbooksDir = (<string>this.args.get("playbooksDir")) || __dirname + "/../playbooks/";
+            let playbookDirs = fs.readdirSync(playbooksDir);
+            for (let index in playbookDirs) {
+                let indexFile = playbooksDir + playbookDirs[index] + "/index.asciidoc";
+                if (fs.existsSync(indexFile)) {
+                    let playbook = parser.parse(indexFile);
+                    playbook.name = playbookDirs[index] + "/";
+                    playbook.path = playbooksDir + playbookDirs[index] + "/";
+                    this.playbooks.push(playbook);
+                }
             }
+        } catch(e) {
+            throw new Error("Error while parsing playbook: " + e);
```

Comment:
Why should that fix the error? You are catching an exception and rethrow it. I guess the error might be suppressed as the function is declared asynchronously.

Reply:
So you should handle the exception in the subscription of the resulting promise

Reply:
This is just to have a more custom error message to let the user know that the error occurred while parsing the playbook. In the run method, this error is caught and pushed to the error array. And as a result of this, the pipeline turns red when the parsing fails.